packages/client/modules/teamDashboard/components/AgendaList/AgendaList.tsx

```diff
@@ -50,18 +50,19 @@ const AgendaList = (props: Props) => {
 
   const onDragEnd = useEventCallback((result) => {
     const {source, destination} = result
-    const destinationItem = agendaItems[destination.index]
-    const sourceItem = agendaItems[source.index]
     if (
-      !destination ||
+      !destination || !source ||
       destination.droppableId !== AGENDA_ITEM ||
       source.droppableId !== AGENDA_ITEM ||
-      destination.index === source.index ||
-      !destinationItem ||
-      !sourceItem
+      destination.index === source.index
```

Comment:
Since we have repeated similar logic three times, should we be able to consider refactoring: to extract a common method e.g. `isInvaildDroppable(result, type)`, pseudo-code below:

```js
const isInvaildDroppable = ({ source, destination }, type) =>  {
     if (!destination) {
       return false
     }
      return destination.droppableId !== type ||
      source.droppableId !== type ||
      destination.index === source.index
}
```

Reply:
I thought about this idea earlier but then you will have more undefined check after this so it doesn't necessarily reduce lots of codes.

Here, you need to check undefined for `destination` so you basically need to move that out of the `isInvaildDroppable` method. Essentially that method is just a wrapper around a if conditional check so 🤷‍♂️.

<img width="543" alt="Screen Shot 2022-02-25 at 4 18 15 PM" src="https://user-images.githubusercontent.com/1879975/155819983-6974da15-28bf-4069-b6e7-345bd6732033.png">
: